### PR TITLE
feat: add indent-spaces option

### PR DESCRIPTION
**Pull request by joshuaprince**
Add "--indent-spaces" integer option to specify the number of spaces to add in the TOC per indentation level. Defaults to the current width of 3 spaces.

Add test cases for the new "indentSpaces" parameter as well as the existing "indentCharacters" parameter.

---

This is a great tool! For my use case, I had to use 2 spaces per indent in the TOC. This seems like the best way to do it. I also added a bonus test case for the flag this one is modeled on. Validated passing tests, and the option shows and works in the browser.

### CODE REVIEW

**Comment by derlin** [comment]
This is a great PR, thank you @joshuaprince !

Two minor fixes before I can merge:
1. can you also update the README (the `Usage: bitdowntoc ...` code block) so it shows the new option?
2. can you rephrase your commit so it uses a present imperative tone? For instance `feat: add indent-spaces option`?

**Comment by joshuaprince** [comment]
Done!

**Review by derlin — APPROVED** [review]
